src/arch/x86/kernel/boot_sys.c

```diff
@@ -63,23 +63,25 @@ BOOT_CODE static paddr_t find_load_paddr(paddr_t min_paddr, word_t image_size)
     int i;
 
     for (i = 0; i < boot_state.mem_p_regs.count; i++) {
-        paddr_t start = MAX(min_paddr, boot_state.mem_p_regs.list[i].start);
+        paddr_t start = MAX(boot_state.ki_p_reg.end, MAX(min_paddr, boot_state.mem_p_regs.list[i].start));
         paddr_t end = boot_state.mem_p_regs.list[i].end;
         word_t region_size = end - start;
 
-        if (region_size >= image_size) {
+        if (end > start && region_size >= image_size) {
```

Comment:
what has changed to necessitate a change here?

Reply:
I think that checks for overflows and is just a bugfix. The first two changes should have their own commit(s) with explanations, they're not part of Xen/HVM support directly.